### DISCUSSION TITLE
Werid git error that is not letting me to push

### DISCUSSION THREAD

**Discussion by thecowmilk** [community/community#22011] [New to GitHub]
<p>Hi guys, yesterday I wanted to merge with a remote repo I needed to <code>commit/push</code>. After that I’d do that I did some <code>merge</code>… Well I tried. Now that it seems that it is up to date but I’m failing to <code>push</code>.</p>
<p>Here is what shows me when i try to push:</p>
<pre data-code-wrap="Enumerating"><code class="lang-nohighlight">Counting objects: 100% (24250/24250), done.
Compressing objects: 100% (16143/16143), done.
Writing objects: 100% (24245/24245), 109.09 MiB | 1.18 MiB/s, done.
Total 24245 (delta 7385), reused 24074 (delta 7288)
remote: Resolving deltas: 100% (7385/7385), completed with 2 local objects.
To https://github.com/masterital/custom-addons
 ! [remote rejected]   12 -&gt; 12 (shallow update not allowed)
error: failed to push some refs to 'https://github.com/masterital/custom-addons'
</code></pre>
<p>When I try the <code>fix</code> to <code>unshallow</code>:</p>
<pre><code class="lang-auto">git fetch --unshallow

remote: fatal: bad object 2fdab87d9d9499c2e0b11aa06ecd33b5cc8e078e
fatal: protocol error: bad pack header
</code></pre>
<p>Can someone help me please?</p>

**Comment by queenofcorgis** [comment]
<p>Nice to see you back in the community <a class="mention" href="/thecowmilk">@thecowmilk</a>! Working in a shallow clone is difficult for these kinds of things, especially if you need to fetch later. Are you just running <code>git push</code> with no arguments? Have you tried <code>git push origin &lt;branch-name&gt;</code>? It might be worth giving that a try.</p>
<p>If you’re still running into this: please share your exact steps including your repo, how you cloned, and if the repo has any force-pushes or similar?</p>

**Comment by thecowmilk** [comment, marked as answer]
<p>Hello <a class="mention" href="/liligalante">@liligalante</a> and thanks for the welcome!</p>
<p>I actually got to fix this issue just by <code>git clone --depth=1</code>. Since I didn’t have the file which was supposed to be cloned I might have had that <code>"shallow clone"</code>. I just had to clone to a recent push which was the files were the same and now everything is okay.</p>
<p>Thanks for your precious time though! &lt;3 hope I’d see you around more! Have a nice day!</p>

**Comment by queenofcorgis** [comment]
<p>Thanks for being so kind and letting us know the solution :heart: Hope to see you around as well, enjoy day!</p>
